### ISSUE TITLE
Should HTTP 400+ be considered errors?

### ISSUE THREAD

**Issue by DavidBruant** [d3/d3-fetch#7]
The current [d3.json](https://github.com/d3/d3-fetch/blob/1aa1f9210e0159a60f578e355c74c9c1a70f1a62/src/json.js) function would not consider a 404 or a 500 an error, but doesn't forward the status code to verify it either. This might lead to hard-to-debug code.

Happy to send a PR if there is consensus

**Comment by mbostock** [comment]
Yes, this is a bug; we should check the response status code. If people want to parse the responses for non-success status codes, it should be reasonable for them to just use the Fetch API directly (since this module is already a very thin layer on top of fetch).

The logic in d3-request was something like:

```js
if (!status && hasResponse(xhr)
    || status >= 200 && status < 300
    || status === 304) {
  …
}
```

I think what we want in d3-fetch is to check [*response*.ok](https://developer.mozilla.org/docs/Web/API/Response/ok).

**Comment by DavidBruant** [comment]
<3

**Comment by mbostock** [comment]
Fixed in 1.0.1. Thanks for the feedback!
